File: src/leetcode/java/RemoveLinkedListElements_203.java
package leetcode.java;

public class RemoveLinkedListElements_203 {
  public ListNode removeElements(ListNode head, int val) {
    if (head == null) return null;
    head.next = removeElements(head.next, val);
    return head.val == val ? head.next : head;
  }

  private ListNode nextNode(ListNode node, int val) {
    if (node == null) return null;
    node.next = nextNode(node.next, val);
    if (node.val == val) return node.next;
    return node;
  }

  class ListNode {
    int val;
    ListNode next;

    ListNode(int x) {
      val = x;
    }
  }
}

File: src/leetcode/python/RemoveLinkedListElements_203.py
class ListNode(object):
    def __init__(self, x):
        self.val = x
        self.next = None


class RemoveLinkedListElements_203(object):
    def removeElements(self, head, val):
        """
        :type head: ListNode
        :type val: int
        :rtype: ListNode
        """
        if not head:
            return head
        p = ListNode(-1)
        p.next = head
        it = p
        while it and it.next:
            if it.next.val == val:
                it.next = it.next.next
            else:
                it = it.next
        return p.next
